driver/sdcard_spi: fix of issue #9667

gschorcht opened the pull request:
Format specifier ```%08lx``` were replaced in ```DEBUG``` by ```PRIu32``` to avoid compilation errors for arguments of type ```uint32_t``` on platforms were ```uint32_t``` is defined to be different from type ```unsigned long```.

Fixes https://github.com/RIOT-OS/RIOT/issues/9667

miri64 commented:
You could use `PRIx32` to keep them hexadecimal ;-). Also, please remove the ESP32 related changes from this PR. We have the same issue on most other platforms when compiled with `clang`, since it is stricter with types and does static code analysis.

miri64 commented:
(the test applications `tests/driver_sdcard_spi` and `tests/driver_io1_explained` also need the same kind of fix at least on my local machine).

miri64 commented:
Fixes https://github.com/RIOT-OS/RIOT/issues/9667

gschorcht commented:
Sorry for the ESP32-related changes :worried: I was pretty sure that I created the branch from the master and not from my ESP32 branch. Anyway. I have rebased the change and replaced ```PRIu32``` with ```PRIx32```, which of course is the correct format specifier. Many Thanks.

miri64 commented:
Cherry-picked bd4661b on top of #9398, to check if that fixes the issues for `sdcard_spi` we encountered there.

miri64 commented:
(I will also check tomorrow, if this makes any problems with `newlib-nano` where some of the `PRI` macro format strings aren't implemented).

miri64 commented:
as stated above however, there are still issues in the tests that *use* this module:

* https://ci.riot-os.org/RIOT-OS/RIOT/9398/6e3557ff490c6655cee77772374b337499f36263/output/compile/tests/driver_io1_xplained/nucleo-f401re:llvm.txt
* https://ci.riot-os.org/RIOT-OS/RIOT/9398/6e3557ff490c6655cee77772374b337499f36263/output/compile/tests/driver_io1_xplained/samr21-xpro:llvm.txt
* https://ci.riot-os.org/RIOT-OS/RIOT/9398/6e3557ff490c6655cee77772374b337499f36263/output/compile/tests/driver_sdcard_spi/nucleo-f401re:llvm.txt
* https://ci.riot-os.org/RIOT-OS/RIOT/9398/6e3557ff490c6655cee77772374b337499f36263/output/compile/tests/driver_sdcard_spi/samr21-xpro:llvm.txt
